filebeat/harvester/log_file.go

```diff
@@ -164,3 +165,10 @@ func (r *LogFile) wait() {
 		}
 	}
 }
+
+func (r *LogFile) Close() {
+	// Make sure reader is only closed once
+	r.singleClose.Do(func() {
+		close(r.done)
```

Comment:
as the underlying reader might block on syscall, the file must be closed right after closing the channel.